core/domain/suggestion_services.py

```diff
@@ -162,7 +172,10 @@ def _update_suggestion(suggestion):
 
     suggestion_model.status = suggestion.status
     suggestion_model.final_reviewer_id = suggestion.final_reviewer_id
-    suggestion_model.change_cmd = suggestion.change_cmd.to_dict()
+    if isinstance(suggestion.change_cmd, dict):
```

Comment:
This looks bad. Shouldn't the format of Suggestion be specified (and be either a dict or not a dict)? Shouldn't branch on the type, like this.

Reply:
Hmmm. I thought so. Shall I make a class QuestionSuggestionCommandObject or something like that, with a to_dict() function and use an object of that class in the suggestion domain object?

Reply:
Ah in SuggestionEditStateContent, it was a domain object (of type ExplorationChange). I was saying I'll add a new domain class for the suggestion for questions, and use that.

Reply:
There should be a domain object of type QuestionChange. Some refactoring may need to happen if there isn't already one.

Reply:
Ok. I'll try that out. However, one issue that could happen is in the apply_change_list function, it takes a question_id along with the change_list. This change cmd which I create should not ideally be used outside of the suggestions model (I think), so shall I prevent it from being used there?

Reply:
Yes, maybe you could check for it at the controller level. I.e. it can be used in the suggestions controller but probably shouldn't be used in the regular question editor controller.

Reply:
Done